cdec/installation-manager-cli-bundle/src/main/java/com/codenvy/im/cli/command/ConfigCommand.java

```diff
@@ -15,54 +15,131 @@
 package com.codenvy.im.cli.command;
 
 
+import com.codenvy.im.artifacts.Artifact;
 import com.codenvy.im.artifacts.CDECArtifact;
+import com.codenvy.im.artifacts.InstallManagerArtifact;
 import com.codenvy.im.managers.Config;
+import com.codenvy.im.managers.PropertyNotFoundException;
 import com.codenvy.im.response.BasicResponse;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.google.common.collect.ImmutableMap;
-
+import org.apache.commons.lang.StringUtils;
+import org.apache.karaf.shell.commands.Argument;
 import org.apache.karaf.shell.commands.Command;
 import org.apache.karaf.shell.commands.Option;
 import org.eclipse.che.commons.json.JsonParseException;
 
 import java.io.IOException;
 import java.util.Map;
+import java.util.TreeMap;
 
 import static com.codenvy.im.artifacts.ArtifactFactory.createArtifact;
 import static com.google.api.client.repackaged.com.google.common.base.Strings.isNullOrEmpty;
+import static java.lang.String.format;
+import static org.fusesource.jansi.Ansi.Color.DEFAULT;
+import static org.fusesource.jansi.Ansi.Color.GREEN;
+import static org.fusesource.jansi.Ansi.Color.WHITE;
+import static org.fusesource.jansi.Ansi.ansi;
 
-/** @author Anatoliy Bazko */
-@Command(scope = "codenvy", name = "im-config", description = "Configure installation manager")
+/**
+ * @author Anatoliy Bazko
+ * @author Dmytro Nochevnov
+ */
+@Command(scope = "codenvy", name = "im-config", description = "Configure Codenvy on-prem")
 public class ConfigCommand extends AbstractIMCommand {
 
-    @Option(name = "--hostname", description = "new Codenvy hostname", required = false)
+    @Argument(index = 0, name = "property", description = "Codenvy property name", required = false, multiValued = false)
+    private String property;
+
+    @Argument(index = 1, name = "value", description = "Codenvy property value", required = false, multiValued = false)
+    private String value;
+
+    @Option(name = "--hostname", description = "New Codenvy hostname", required = false)
     private String hostname;
 
+    @Option(name = "--im-cli", description = "To get installation manager CLI properties", required = false)
+    private boolean imCli;
+
+    private Artifact cdecArtifact;
+
     /** {@inheritDoc} */
     @Override
     protected void doExecuteCommand() throws Exception {
+        cdecArtifact = createArtifact(CDECArtifact.NAME);
+
+        if (imCli) {
+            doGetConfig(createArtifact(InstallManagerArtifact.NAME));
+            return;
+        }
+
         if (!isNullOrEmpty(hostname)) {
             doUpdateCodenvyHostUrl();
             return;
```

Comment:
I would prefer to have atomic operations, to simplify output and error handling.